src/structs.h

```diff
@@ -233,20 +233,31 @@ struct room_data
   struct obj_data *best_workshop[NUM_WORKSHOP_TYPES];
   
 #ifdef USE_DEBUG_CANARIES
-  // No sense in initializing the value since it's memset to 0 in most invocations.
   int canary;
 #endif
   room_data() :
-      name(NULL), description(NULL), night_desc(NULL), ex_description(NULL),
-      matrix(0), access(0), io(0), trace(0), bandwidth(0), jacknumber(0), 
-      address(NULL), peaceful(0), func(NULL), dirty_bit(FALSE),
-      staff_level_lock(0), contents(NULL), people(NULL), vehicles(NULL), watching(NULL)
+      number(0), zone(0), sector_type(0), name(NULL), description(NULL), night_desc(NULL), ex_description(NULL),
+      matrix(0), access(0), io(0), trace(0), bandwidth(0), rtg(0), jacknumber(0), address(NULL), room_flags(0),
+      blood(0), debris(0), spec(0), rating(0), cover(0), crowd(0), type(0), x(0), y(0), z(0), peaceful(0), func(NULL),
+      dirty_bit(FALSE), staff_level_lock(0), elevator_number(0), contents(NULL), people(NULL), vehicles(NULL),
+#ifdef USE_DEBUG_CANARIES
+      canary(0),
+#endif
+      watching(NULL)
   {
     for (int i = 0; i < NUM_OF_DIRS; i++) {
       dir_option[i] = NULL;
       temporary_stored_exit[i] = NULL;
     }
     
+    memset(vision, 0, 3 * sizeof(int));
```

Comment:
It might be better to do `memset(vision, 0, sizeof(vision) * sizeof(vision[0]));` (provided that sizeof works like I think it does and gives the number of elements in the array when given an array like that). This will prevent us from having to come back and change the memset if we change the variable type later.

Reply:
No sizeof(vision[0]) will return the size of the pointer to the value (so a fixed size depending on platform). All arrays used as parameters to function calls as pointers or by value, decay into pointers in C. What is actually passed to the function is a pointer to the value. Look up "array decay in C++" for detailed info.